Scoop/Controller/Home/HomeViewController.swift

```diff
@@ -0,0 +1,220 @@
+        tableView.register(HomeTableViewCell.self, forCellReuseIdentifier: homeCellIdenitifer)
+        view.addSubview(tableView)
+        
+        tableView.snp.makeConstraints { make in
+            make.top.equalTo(headerView.snp.bottom)
+            make.leading.trailing.equalToSuperview()
+            make.bottom.equalTo(view.safeAreaLayoutGuide)
+        }
+    }
+    
+    private func setupSignOutButton() {
+        signOutButton.setTitle("Sign Out", for: .normal)
+        signOutButton.setTitleColor(.white, for: .normal)
+        signOutButton.titleLabel?.font = .systemFont(ofSize: 18, weight: .bold)
+        view.addSubview(signOutButton)
+        
+        signOutButton.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+        
+        let signOutAction = UIAction { _ in
+            self.signOut()
+        }
+        
+        signOutButton.addAction(signOutAction, for: .touchUpInside)
+    }
+    
+    private func signOut() {
+        GIDSignIn.sharedInstance.signOut()
+        dismiss(animated: true)
+    }
+    
+    private func getRides() {
+        // Populate table view with dummy data
+        let readesRide = Ride()
+        readesRide.organizer.name = "Reade"
+        readesRide.departureLocation = "Ithaca, NY"
+        readesRide.arrivalLocation = "Darien, CT"
+        readesRide.date = "Mar 3"
+        readesRide.isActive = true
+        
+        let annesRide = Ride()
+        annesRide.organizer.name = "Anne"
+        annesRide.departureLocation = "Ithaca, NY"
+        annesRide.arrivalLocation = "Orlando, FL"
+        annesRide.date = "Apr 10"
+        annesRide.isActive = true
+        
+        let karlsRide = Ride()
+        karlsRide.organizer.name = "Karl"
+        karlsRide.departureLocation = "Ithaca, NY"
+        karlsRide.arrivalLocation = "Pittsburgh, PA"
+        karlsRide.date = "May 13"
+        karlsRide.isActive = true
+        
+        let sarahsRide = Ride()
+        sarahsRide.organizer.name = "Sarah"
+        sarahsRide.departureLocation = "Ithaca, NY"
+        sarahsRide.arrivalLocation = "Montreal, CA"
+        sarahsRide.date = "May 20"
+        sarahsRide.isActive = true
+        
+        activeRides = [readesRide, annesRide]
+        pendingRides = [karlsRide, sarahsRide]
+    }
+    
+}
+
+// MARK: - UITableViewDelegate
+extension HomeViewController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return TableSection.allCases[section].rawValue
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let titleLabel = UILabel()
+        titleLabel.font = .systemFont(ofSize: 22, weight: .regular)
+        titleLabel.text = self.tableView(tableView, titleForHeaderInSection: section)
+        
+        let headerView = UIView()
+        headerView.addSubview(titleLabel)
+        
```

Comment:
can we extract this headerview into it's own file?

Reply:
Sure!